src/sentry/tasks/store.py

```diff
@@ -381,13 +405,18 @@ def _continue_to_save_event():
     return _continue_to_save_event()
 
 
-@instrumented_task(
+@instrumented_task(  # type: ignore
     name="sentry.tasks.store.process_event",
     queue="events.process_event",
     time_limit=65,
     soft_time_limit=60,
 )
-def process_event(cache_key, start_time=None, event_id=None, data_has_changed=None, **kwargs):
+def process_event(
+    cache_key: Optional[str],
+    start_time: Optional[int] = None,
+    event_id: Optional[str] = None,
+    data_has_changed: bool = False,
```

Comment:
I don't think removing `**kwargs` is safe here. Do we have any invocations of `process_event` that pass more arguments? If yes they need to be removed in a separate PR and deployed separately.